Sharing and using the forum (#15)
tiosgz/modding-vivaldi: community/sharing.md

> # Sharing your Mods

Reviewer: Missing recommended forum post content – description, screenshot etc

Author: Yeah, I was going to add screenshot yesterday but forgot. What GIF program do you recommend for Linux?

Reviewer: I have ffmpeg, but that’s probably not what most people are gonna use. I have no idea, I only use text-interface tools for a lot of things.

Author: Well does it export GIF?

Reviewer: It can handle a lot of file types, including gif.

Author: Ok, I will mention it then.

```
/**
 * <TITLE>
	- - -   
 * Description:        <description>
 * Filename:          <filename>
 * Platform:          <Linux/Mac/Windows/All>
 * Version:           Vivaldi version <version>
 * Author(s):          <github username, forum username, other contributors>
 * Topic:             https://forum.vivaldi.net/topic/<topic>
**/
```

### Versioning Your Mods

Instead of tracking each edit to your mod as a separate version, its easier and sometimes more helpful to include the latest version of Vivaldi that it was tested in. Along with the last update time, this will let you and others keep track of the mod version.

Because the forum and Git keep track of posting times, it is probably not necessary to include the last update time in the code header (but you can if you want).

If you do want to version your mods, it would look like this:
```
Version: 2.1 for Vivaldi 3.9.2289.3
```

If you want to keep it simple and use the last vivaldi version, it would look like this (see the header template above):
```
* Version:           Vivaldi 3.9.2289.3
```

> ## Tagging your Mods

Reviewer: Since the rest of the file is about sharing in general, could you make it more clear that you mean tagging forum topics?

# Sharing your Mods
Author(s): code3z (@code3 on the forum)

Once you've created a mod that you're proud of, the next step is to share it.
Sharing will be a big help to other people. You can also get feedback and help

> from the community. But it may also mean you have to respond to bugs in your

Reviewer: ```suggestion
from the community. But it will also mean you have to respond to bugs in your
```

code and help other people use your code.

## Managing and Sharing Mods using Git

> You can use hosted Git repos such as GitHub and Gitlab to upload, manage, share,

Reviewer: *Git repo(sitory) hosting services*. And GitLab also has two uppercase letters in its name.

> and collaborate on mods. See the Managing Mods section of this guide for more

Reviewer: *managing mods guide*

…and I don’t plan to explain much Git there, just mention it & maybe link a tutorial.

Author: Well, maybe mention briefly how to work with it - not actually what things to click in the GUI or commands to type in the CLI, just why you should use it and some modding-specific things about Git.

information on Git.

## Code headers

Putting a comment at the top of your code helps people understand it,

> use it, and remember where it came from. Consider using this template:

Reviewer: *Consider using a header like this* seems better (‘use exactly this’ vs. ‘try to mention the things mentioned here’)

Author: Not clear what is being requested here.

Reviewer: ```suggestion
use it, and remember where it came from. Consider using a header like this:
```

> ```Javascript OR CSS

Reviewer: Invalid syntax spec.

Author: Is there a way to do two languages or just pick one?

Reviewer: Just pick one.

/**
 * <TITLE>
 * https://forum.vivaldi.net/topic/<topic>

> 	- - -   

Reviewer: ```suggestion
```

 * Description:        <description>
 *      <If your description is long, put the second part here>
 * Filename:          <filename>
 * Platform:          <Linux/Mac/Windows/All>
 * Version:           Vivaldi version <version>

>  * Author(s):          <Here you can add your forum username, and username for github or any relevant app (also add any other contributors here)>

Reviewer: I would prefer it like this 🤷
```
 * Description: <description>
 *              <if it is long, continue it here>
 * Version:     for Vivaldi <version>
 * Author:      <forum/GitHub username, contributors>
```

**/
```

### Versioning Your Mods

> Instead of tracking each edit to your mod as a separate version, it's

Reviewer: ```suggestion
Instead of versioning your mods separately, it’s
```

easier and sometimes more helpful to include the latest version of Vivaldi that
it was tested in. Along with the last update time, this will let you and others
know how up-to-date the mod is.

Because both the forum and Git keep track of posting times, it's not necessary

> to include the last update time in the code header (but you can if you want).

Reviewer: Is this paragraph needed?

> If you do want to version your mods, it would look like this:

Reviewer: ```suggestion
If you do want to version your mods, please include the Vivaldi version as well.
```
…and then I think you can omit everything up to the next heading.

```
 * Version: 2.1 for Vivaldi 3.9.2289.3
```

If you want to keep it simple and use the last Vivaldi version, it would look
like this (see the header template above):

```
 * Version:           Vivaldi version 3.9.2289.3
```

> ## Tagging Forum Posts

Reviewer: Could you put this & the following headings under `## Posting Mods to Forum`? (& thus also increase their heading level to 3)

Author: Got it

Tagging your mods helps others find them. Use the forum tagging feature to
insert the following tags, in addition to any others that make sense:

>  1. [Mod]

Reviewer: `modding` seems more common than `mod` (if you update this, consider updating the screenshot as well)

>  2. [CSS] and/or [JS]

Reviewer: The tag is `javascript`

 3. [Area] (what area of the UI does it affect?)

> > If the tag doesn't exist the forum software will create it  -

Reviewer: ```suggestion
If the tag doesn't exist, the forum software will create it—
```

but please don't make new tags on your own!
Only use tags that are suggested from the drop-down.

![Tagging Posts Screenshot](../assets/screenshots/tagging-posts.png)

## Describing your mods

Now that you've added your code, titled your post, and tagged it,
you need to write content! What you write is really up to you, but be sure to
include what the mod does, how it will help people, and if it has any
shortcomings or bugs or if it does not work with certain settings.

You should also **include a screenshot**. If the mod does something that can only
be shown through a screen recording, take a screen recording and upload it as a GIF.
You can find some screen recording tools in the resources section of this guide.

Good job, I hope to see your mods soon! 👍

## Editing Posts

To edit a post and update the code, click the three-dot menu. Alternatively, if

> using Git or cloud storage, just link to the file in the Git repository and

Reviewer: ```suggestion
using Git or cloud storage, just link to the file and
```
(coz you mentioned cloud storage)

Also, I think you should mention it earlier than in *editing* posts.

Once you've created a mod that you're proud of, the next step is to share it. Sharing will be a big help to other people. You can also get feedback and help from the community. But it may also mean you have to respond to bugs in your code and help other people use your code.

> ## Managing Mods

Reviewer: Not sure if this is the best way to express that you can use online-hosted git repos 🤷

 * https://forum.vivaldi.net/topic/<topic>
	- - -   
 * Description:        <description>

>  <If your description is long, put the second part here>

Reviewer: ```plaintext
 <If your description is long, put the second part here>
 not here
 *                    but here
```

Author: Isn't the (*) for a new line?

Author: I mean a new piece of content

Reviewer: No, the asterisk if for *any* new line inside the comment.

Reviewer: ```plaintext
 * I mean, not even here
 *                    but here
```
